common/Windows/SecondaryWindow.cs

```diff
@@ -0,0 +1,328 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+using DevHome.Common.Extensions;
+using DevHome.Common.Services;
+using DevHome.Contracts.Services;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Markup;
+using Windows.Win32;
+using Windows.Win32.Foundation;
+using Windows.Win32.UI.WindowsAndMessaging;
+using WinUIEx;
+
+namespace DevHome.Common.Windows;
+
+[ContentProperty(Name = nameof(SecondaryWindowContent))]
+public class SecondaryWindow : WindowEx
+{
+    private readonly SecondaryWindowTemplate _windowTemplate;
+    private WindowEx? _primaryWindow;
+    private bool _useAppTheme;
+    private bool _isModal;
+    private bool _isTopLevel;
+
+    private WindowEx MainWindow => Application.Current.GetService<WindowEx>();
+
+    private IThemeSelectorService ThemeSelector => Application.Current.GetService<IThemeSelectorService>();
+
+    private IAppInfoService AppInfo => Application.Current.GetService<IAppInfoService>();
+
+    public WindowTitleBar? WindowTitleBar
+    {
+        get => _windowTemplate.TitleBar;
+        set
+        {
+            if (WindowTitleBar != value)
+            {
+                // Remove title changed event handler from previous title bar
+                if (WindowTitleBar != null)
+                {
+                    WindowTitleBar.TitleChanged -= OnSecondaryWindowTitleChanged;
+                }
+
+                // Set new title bar
+                _windowTemplate.TitleBar = value;
+
+                // By default, if no title is set, use the application name as title
+                var title = value?.Title;
+                title = string.IsNullOrEmpty(title) ? AppInfo.GetAppNameLocalized() : title;
+                OnSecondaryWindowTitleChanged(null, title);
+
+                // Add title changed event handler to new title bar
+                if (value != null)
+                {
+                    value.Title = title;
+                    value.TitleChanged += OnSecondaryWindowTitleChanged;
+                }
+            }
+    }
+    }
+
+    /// <summary>
+    /// Gets or sets the window content in the custom layout.
+    /// </summary>
+    /// <remarks>
+    /// This is the default content of the secondary window.
+    /// See also <seealso cref="ContentPropertyAttribute"/>.
+    /// </remarks>
+    public object SecondaryWindowContent
```

Comment:
This property is inherently an `object` because it sets the `Content` of a `ContentControl` inside the `SecondaryWindowTemplate`. The `ContentControl.Content` takes an `object` type allowing `UIElements`, but also some primitive types as text and numbers. So all the below example should work with type `object`:

```xml
<SecondaryWindow>
    <TextBlock Text="Hello world" />
</SecondaryWindow>
<SecondaryWindow>42</SecondaryWindow>
<SecondaryWindow>Hello world</SecondaryWindow>
```